internal/loader/typed_setting.go

```diff
@@ -118,3 +116,80 @@ func isJsonContentType(contentType *string) bool {
 	matched, _ := regexp.MatchString("^application\\/(?:[^\\/]+\\+)?json(;.*)?$", contentTypeStr)
 	return matched
 }
+
+func UnmarshalConfigMap(existingConfigMapSetting *map[string]string, dataOptions *acpv1.ConfigMapDataOptions) (interface{}, map[string]interface{}, error) {
+	allSettings := make(map[string]interface{})
+	existingConfigMapData := (*existingConfigMapSetting)[dataOptions.Key]
+	var typedFeatureFlags interface{}
+	var err error
+
+	switch dataOptions.Type {
+	case acpv1.Yaml:
```

Comment:
Please comments that type has no possibility of being `default` or `properties` here